src/plugin.rs

```diff
@@ -0,0 +1,69 @@
+use libloading::Library;
+use once_cell::sync::OnceCell;
+use std::ffi::{CString, OsStr};
+use std::ptr;
+use tree_sitter::Node;
+
+#[cfg(unix)]
+type RawSymbol<TValue> = libloading::os::unix::Symbol<TValue>;
+#[cfg(windows)]
+type RawSymbol<TValue> = libloading::os::windows::Symbol<TValue>;
+
+pub struct Filterer {
+    // filterer: Symbol<fn(&Node) -> bool>,
+    filterer: RawSymbol<unsafe extern "C" fn(*const Node) -> bool>,
+    _library: Library,
+}
+
+impl Filterer {
+    pub fn call(&self, node: &Node) -> bool {
+        unsafe { (self.filterer)(node) }
+    }
+}
+
+fn load_plugin(library_path: impl AsRef<OsStr>, filter_arg: Option<&str>) -> Filterer {
+    let library =
+        unsafe { Library::new(library_path).expect("Couldn't load filter dynamic library") };
+
+    if let Some(initialize) =
+        unsafe { library.get::<unsafe extern "C" fn(*const libc::c_char)>(b"initialize\0") }.ok()
```

Comment:
Ok so the plugin "API" I went with is:
Optional `initialize()` function that takes a C-style string (so plugins that don't need to be "configured" by the `--filter-arg` command-line argument can omit `initialize()` entirely)
Specifically if no `--filter-arg` argument is passed but `initialize()` is implemented then it gets passed a null pointer

`filterer()` function that takes what should effectively be a pointer to a C tree-sitter `Node` and returns a bool

Reply:
Woof ok I know that the expected type is `&[u8]` so I guess `b"..."` ("byte strings"?) gives you an `&[u8]` rather than a `&str`?

The `\0` I saw in some examples (but not others) and then saw in the `libloading` docs that "Providing a null-terminated `symbol` may help to avoid an allocation."

Ok yes looks like it's basically [ASCII-encoding](https://doc.rust-lang.org/reference/tokens.html#byte-string-literals)